src/main/java/hexgui/util/Options.java

```diff
@@ -0,0 +1,294 @@
+package hexgui.util;
+
+import java.io.*;
+import java.util.*;
+
+/** Parser for command line options. Options begin with a single '-' character. */
+public class Options {
+  /**
+   * Parse options.
+   *
+   * @param args Command line args from main method
+   * @param specs Specification of allowed options. Contains option names (without '-'). Options
+   *     that need an argument must have a ':' appended. The special argument '--' stops option
+   *     parsing, all following arguments are treated as non-option arguments.
+   * @throws Exception If options are not valid according to specs.
+   */
+  public Options(String[] args, String[] specs) throws Exception {
+    for (String spec : specs) {
+      if (spec.length() > 0) m_map.put(spec, null);
+    }
+    parseArgs(args);
+  }
+
+  /** Check if option is present. */
+  public boolean contains(String option) {
+    String value = get(option, null);
+    return (value != null);
+  }
+
+  /**
+   * Return string option value.
+   *
+   * @param option The option key.
+   * @return The option value or an empty string, if option is not present.
+   */
+  public String get(String option) {
+    return get(option, "");
+  }
+
+  /**
+   * Return string option value.
+   *
+   * @param option The option key.
+   * @param defaultValue The default value.
+   * @return The option value or defaultValue, if option is not present.
+   */
+  public String get(String option, String defaultValue) {
+    assert isValidOption(option);
+    String value = getValue(option);
+    if (value == null) return defaultValue;
+    return value;
+  }
+
+  /**
+   * Get remaining arguments that are not options.
+   *
+   * @return The sequence of non-option arguments.
+   */
+  public ArrayList<String> getArguments() {
+    return m_args;
+  }
+
+  /**
+   * Check that the number of non-option arguments is zero.
+   *
+   * @throws Exception If there are any non-option arguments.
+   */
+  public void checkNoArguments() throws Exception {
+    if (!m_args.isEmpty())
+      throw new Exception("Command does not allow arguments that are not options");
+  }
+
+  /**
+   * Parse double option.
+   *
+   * @param option The option key.
+   * @return The option value or 0, if option is not present.
+   * @throws Exception If option value is not a double.
+   */
+  public double getDouble(String option) throws Exception {
+    return getDouble(option, 0);
+  }
+
+  /**
+   * Parse double option.
+   *
+   * @param option The option key.
+   * @param defaultValue The default value.
+   * @return The option value or defaultValue, if option is not present.
+   * @throws Exception If option value is not a double.
+   */
+  public double getDouble(String option, double defaultValue) throws Exception {
+    String value = get(option, Double.toString(defaultValue));
+    if (value == null) return defaultValue;
+    try {
+      return Double.parseDouble(value);
+    } catch (NumberFormatException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)